File: apps/api-kotlin/src/main/kotlin/com/couplecalendar/application/dto/response/AuthResponse.kt
package com.couplecalendar.application.dto.response

data class AuthResponse(
    val accessToken: String,
    val refreshToken: String? = null,
    val user: UserInfo
) {
    data class UserInfo(
        val id: String,
        val email: String,
        val nickname: String
    )
}

/**
 * 토큰 갱신 응답. accessToken 만료 시 refresh token으로 발급받은 새 토큰 쌍.
 * (Supabase는 refresh token도 회전시키므로 새 refreshToken을 함께 저장해야 한다.)
 */
data class TokenRefreshResponse(
    val accessToken: String,
    val refreshToken: String? = null
)

File: apps/web/src/data/dto/auth-response.ts
export interface AuthUserInfo {
	id: string;
	email: string;
	nickname: string;
}

export interface AuthResponse {
	accessToken: string;
	/** accessToken 만료 시 갱신에 쓰는 refresh token. 백엔드(Supabase)가 함께 내려준다. */
	refreshToken?: string;
	user: AuthUserInfo;
}
